components/omega/doc/design/ShallowWaterOmega0.md

```diff
@@ -0,0 +1,212 @@
+
+The manufactured solution test provides an exact solution in time for the full nonlinear shallow water equations (momentum and thickness). It tests the time stepping scheme along with the the full advection terms and the SSH gradient.  It is conducted on a doubly periodic cartesian mesh, so does not test boundary conditions. Error should converge at 1st order, as described by [Bishnu 2021](https://zenodo.org/record/7439539) and Bishnu et al. 2023. Also see the manufactured solution test case in Polaris.
+
+Requirements for tests are:
+
+- expected order of convergence
+- conservation of total volume
+- automation and reproducibility in polaris
+
+This test should be conducted as soon as momentum and thickness equations and time-stepping is in place. It does not require any tracer infrastructure.
+
+### 5.4 Advection of cosine bell on a sphere
+
+Advection of a tracer cosine bell around the sphere was described in [Williamson et al. 1992](https://www.sciencedirect.com/science/article/pii/S0021999105800166) as test case 1. It tests the tracer advection and tracer time stepping. Velocity and thickness are predefined fields and remain fixed, so this does not exercise those modules.
+
+Requirements for tests are:
+
+- expected order of convergence (check)
+- conservation of total tracer amount
+- tracer min and max values should remain bounded by initial bounds.
+- automation and reproducibility in polaris
+
+### 5.5 Performance testing
+
+Tests can be conducted with inertia-gravity wave tests but with full non-linear terms and 100 identical layers. Domain will be cartesian resolutions from 64x64 up to 512x512. These can be compared with Bishnu 2023 (ref). Additionally, we could test spherical cases with a Williamson test case or similar.
+
+Requirements for tests are:
+
+- Single CPU performance should be as good or better than single CPU MPAS-Ocean.
+- GPU tests on single node should be better than single node CPU tests.
+- Scaling on CPUs multi-node, up to 4096 cores or higher, should be close to perfect scaling for 512x512 mesh, and as good or better than MPAS-Ocean.
+- Scaling on GPUs for multi-node should be close to perfect scaling for 512x512 mesh.
+- automation and reproducibility in polaris
+
+### 5.6 Demonstration cases with realistic coastlines
+
+Tests in realistic domains will facilitate the testing of boundary conditions on realistic coastlines, which are not included in any of the previous tests. They will also be useful for highlights and talks to show that Omega-0 can run in global realistic domains, albeit with shallow water equations only.
+
+Potential test cases include:
+
+1. Coastal inundation: Delaware Bay mesh, after ICOM tests, to demonstrate boundary conditions, resolution, and for 'splashy' images. Would tidal forcing be required for that?
+1. Global realistic Tsunami to test boundary conditions and grid-scale noise. See Darren Engwirda's recent tests on Alaska's Aleutian Islands.
```

Comment:
Don't (1) and (2) require the unified earth system mesh + wetting and drying from ICoM?  I'm pretty concerned about listing these as possible tests.